tests/pipelines/test_pipelines_text_generation.py

```diff
@@ -477,3 +479,19 @@ def test_pipeline_length_setting_warning(self):
         with CaptureLogger(logger) as cl:
             _ = text_generator(prompt, max_length=10)
         self.assertNotIn(logger_msg, cl.out)
+
+    @require_torch
+    def test_pipeline_tokenizer_has_pad_but_model_doesnt(self):
+        # When the tokenizer pad_token_id is set but the model pad_token_id is not, we pass the pad_token_id to
+        # `generate`. This prevents a warning from being raised, which this test checks.
+        model = AutoModelForCausalLM.from_pretrained("hf-internal-testing/tiny-random-gpt2")
+        tokenizer = AutoTokenizer.from_pretrained("hf-internal-testing/tiny-random-gpt2")
+        model.config.pad_token_id = None
+        model.generation_config.pad_token_id = None
+        tokenizer.pad_token_id = tokenizer.eos_token_id
+
+        llm = pipeline(task="text-generation", model=model, tokenizer=tokenizer, framework="pt")
+        with self.assertRaises(AssertionError) as exc:
+            with self.assertLogs("transformers", level="WARNING"):
+                llm("The capital of France ")
+        self.assertIn("no logs of level WARNING or higher triggered", str(exc.exception))
```

Comment:
This is a very flaky test - essentially if anyone adds a new check which might legitimately raise a warning this is going to fail. 

Admittedly, we should try to avoid having any warnings trigger in our tests (we should run things properly) but it's going to be pretty annoying to try and handle.

I'd rather we just didn't add this test tbh